desktop-widgets/mainwindow.cpp

```diff
@@ -1876,13 +1875,7 @@ void MainWindow::on_paste_triggered()
 
 void MainWindow::on_actionFilterTags_triggered()
 {
-	if (ui.multiFilter->isVisible()) {
-		ui.multiFilter->closeFilter();
-		ui.actionFilterTags->setChecked(false);
-	} else {
-		ui.multiFilter->setVisible(true);
-		ui.actionFilterTags->setChecked(true);
-	}
+	setApplicationState(getCurrentAppState() == "FilterDive" ? "Default" : "FilterDive");
```

Comment:
not to quibble, but the text of the commit message (sorry, can't comment on commit messages, so commenting in the code) of "dropping tons of code" seems like a bit of an overstatement?